nifi-extension-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/GetS3ObjectMetadata.java

```diff
@@ -166,14 +168,22 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
                 Map<String, Object> combinedMetadata = new HashMap<>(metadata.getRawMetadata());
                 combinedMetadata.putAll(metadata.getUserMetadata());
 
-                if (!isRouter && context.getProperty(METADATA_TARGET).getValue().equals(TARGET_ATTRIBUTES.getValue())) {
+                if (context.getProperty(METADATA_TARGET).getValue().equals(TARGET_ATTRIBUTES.getValue())) {
                     String attributePrefix = context.getProperty(METADATA_ATTRIBUTE_PREFIX).getValue();
                     Map<String, String> newAttributes = combinedMetadata
                             .entrySet().stream()
+                            .filter(e -> {
+                                if (attributePattern == null) {
+                                    return true;
+                                } else {
+                                    return attributePattern.matcher(attributePrefix + e.getKey())
+                                            .find();
```

Comment:
It seems like the match should only consider `e.getKey()`, and the prefix should be added after, as it is now.